src/neo4j_graphrag/experimental/pipeline/pipeline.py

```diff
@@ -224,13 +291,30 @@ def get_pygraphviz_graph(self, hide_unused_outputs: bool = True) -> pgv.AGraph:
                     source_output_node = f"{source_component}.{source_param_name}"
                 else:
                     source_output_node = source_component
-                G.add_edge(source_output_node, component_name, label=param)
-        # remove outputs that are not mapped
+
+                if source_output_node in node_ids and component_name in node_ids:
+                    relationships.append(
+                        Relationship(
+                            source=node_ids[source_output_node],
+                            target=node_ids[component_name],
+                            caption=param,
+                            color="#EA4335",  # Red for parameter connections
+                            caption_align=CaptionAlignment.CENTER,
+                            caption_size=10,
+                        )
+                    )
+
+        # Filter unused outputs if requested
         if hide_unused_outputs:
-            for n in G.nodes():
-                if n.attr["node_type"] == "output" and G.out_degree(n) == 0:  # type: ignore
-                    G.remove_node(n)
-        return G
+            used_nodes = set()
+            for rel in relationships:
```

Comment:
I think there is a logic issue here: we do not want to filter out isolated nodes (this should not happen I think?), but node of type "output" with no outgoing relationship.